src/main/ts/components/Editor.tsx

```diff
@@ -310,194 +230,310 @@ export class Editor extends React.Component<IAllProps> {
-          this.props.init.init_instance_callback(editor);
+        if (props.init && isFunction(props.init.init_instance_callback)) {
+          props.init.init_instance_callback(editor);
         }
       }
     };
-    if (!this.inline) {
+    if (!inline) {
       target.style.visibility = '';
     }
     if (isTextareaOrInput(target)) {
-      target.value = this.getInitialValue();
+      target.value = getInitialValue();
     }
 
     tinymce.init(finalInit);
-  };
-}
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  React.useEffect(() => {
+    if (getTinymce(view) !== null) {
+      initialise();
+    } else if (Array.isArray(tinymceScriptSrc) && tinymceScriptSrc.length === 0) {
+      onScriptsLoadError?.(new Error('No `tinymce` global is present but the `tinymceScriptSrc` prop was an empty array.'));
+    } else if (elementRef.current?.ownerDocument) {
+      const successHandler = () => {
+        onScriptsLoad?.();
+        initialise();
+      };
+      const errorHandler = (err: unknown) => {
+        onScriptsLoadError?.(err);
+      };
+      ScriptLoader.loadList(
+        elementRef.current.ownerDocument,
+        getScriptSources(),
+        props.scriptLoading?.delay ?? 0,
+        successHandler,
+        errorHandler
+      );
+    }
+
+    const boundHandlers = boundHandlersRef.current;
+
+    return () => {
+      const editor = editorRef.current;
+      if (editor) {
+        editor.off(changeEvents(), handleEditorChange);
+        editor.off(beforeInputEvent(), handleBeforeInput);
+        editor.off('keypress', handleEditorChangeSpecial);
+        editor.off('keydown', handleBeforeInputSpecial);
+        editor.off('NewBlock', handleEditorChange);
+        Object.keys(boundHandlers).forEach((eventName) => {
+          editor.off(eventName, boundHandlers[eventName]);
+        });
+        boundHandlersRef.current = {};
+        editor.remove();
+        editorRef.current = undefined;
+      }
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  React.useEffect(() => {
+    if (editorRef.current !== undefined) {
+      // check if we should monitor editor changes
+      const isValueControlled = onEditorChange !== undefined || value !== undefined;
+      if (isValueControlled) {
+        editorRef.current.on(changeEvents(), handleEditorChange);
+        editorRef.current.on(beforeInputEvent(), handleBeforeInput);
+        editorRef.current.on('keydown', handleBeforeInputSpecial);
+        editorRef.current.on('keyup', handleEditorChangeSpecial);
+        editorRef.current.on('NewBlock', handleEditorChange);
+      } else {
+        editorRef.current.off(changeEvents(), handleEditorChange);
+        editorRef.current.off(beforeInputEvent(), handleBeforeInput);
+        editorRef.current.off('keydown', handleBeforeInputSpecial);
+        editorRef.current.off('keyup', handleEditorChangeSpecial);
+        editorRef.current.off('NewBlock', handleEditorChange);
```

Comment:
I suspect these `off` statements will not work because the references to the functions will have changed before `off` is run. For example `handleEditorChange` is dependent on `onEditorChange` so if the user deregisters that prop it will update the function `handleEditorChange` points to before it tries to deregister the event handlers.

Maybe this is possible with a cleanup on the callback constructors? Be careful because leaving these registered will have very strange results.